README.md

```diff
@@ -64,6 +64,35 @@ Generate coverage reports for the whole project:
 Generate coverage for a single module, i.e.: 
 
     ./gradlew clients:reportCoverage
+
+### Generating an API compatibility report between trunk and your local branch ###
+*Note that the branches need to be local.*
+
+    ./gradlew apiCompatibilityReport
+
+By default it just lists the incompatible changes. If you want a more complete report, you can set the `onlyIncompatible`
+flag to false:
+
+    ./gradlew apiCompatibilityReport -PonlyIncompatible=false
+
+There are some other options too. If you need to control the build result on binary or source incompatible changes, 
+then you can do it so with the following flags. The below example would result in failing only in case of source
+incompatibility.
+
+    ./gradlew apiCompatibilityReport -PfailOnBinaryIncompatibility=false -PfailOnSourceIncompatibility=true
+
+It is often a requirement to control the build outcome based on version differences. Usually it is allowed to do
```

Comment:
We should be specific about what we are allowed to do in Kafka, not talk about what other projects "usually do".  In Kafka we allow incompatible change between major releases, but not minor releases.

Reply:
Fair point, will fix.